django/contrib/auth/hashers.py

```diff
@@ -512,6 +513,84 @@ class BCryptPasswordHasher(BCryptSHA256PasswordHasher):
     digest = None
 
 
+class ScryptPasswordHasher(BasePasswordHasher):
+    """
+    Secure password hashing using the Scrypt algorithm
+
+    The result is a 64 byte binary string.
+    """
+    algorithm = "scrypt"
+    work_factor = 16384
```

Comment:
Do we have any plan on how & when to upgrade those as time passes by? (ie like we have for pbkdf2)

Reply:
I'm not sure, I don't think there is a need to do this per release. :thinking: 

`work_factor` must be a power of two and the OpenSSL limits memory usage to 32 MiB, so we would have to increase `maxmem` as well: :teacher: 
```
(2 ** 15) * (2 * 8) * 64 = 33554432 = 32 MiB
```